src/lib/allocation_tracker.cc

```diff
@@ -265,18 +265,21 @@ DDRes AllocationTracker::push_sample(uint64_t allocated_size,
 
   if (tl_state.stack_end == nullptr) {
     // This call should only occur on main thread
-    tl_state.stack_end = retrieve_stack_end_address();
+    if (retrieve_stack_bounds(tl_state.stack_start, tl_state.stack_end) < 0) {
+      DDRES_RETURN_ERROR_LOG(DD_WHAT_PERFRB, "Unable to get thread bounds");
+    }
   }
 
   event->sample_id.pid = _state.pid;
   event->sample_id.tid = tl_state.tid;
   event->period = allocated_size;
   event->size = PERF_SAMPLE_STACK_SIZE;
 
-  event->dyn_size = save_context(tl_state.stack_end, event->regs,
-                                 ddprof::Buffer{event->data, event->size});
-
-  if (writer.commit(buffer) || notify_consumer) {
+  event->dyn_size =
+      save_context(tl_state.stack_start, tl_state.stack_end, event->regs,
+                   ddprof::Buffer{event->data, event->size});
+  // discard if dyn_size == 0, this way, we will skip over it
```

Comment:
We could consider some form of lost samples. I'm not how meaningful (or confusing) it would be.

Reply:
I agree it would be nice to have some stats on this

Reply:
We do have information, we have the size of the allocation. 
Should we add a frame saying that we could not capture the buffer ? 

`malloc();[broken]`

Reply:
This adds another new form of stack which might be confusing for the user. However this exposes the limitation and makes sure we can explain the lack of visibility. What do you think about this name ? I will have to add a new type of event.

Reply:
We could reuse the `[incomplete]` virtual frame